src/Illuminate/Http/Client/PendingRequest.php

```diff
@@ -1035,10 +1036,24 @@ protected function sendRequest(string $method, string $url, array $options = [])
             $this->transferStats = $transferStats;
         };
 
-        return $this->buildClient()->$clientMethod($method, $url, $this->mergeOptions([
+        $mergedOptions = $this->mergeOptions([
             'laravel_data' => $laravelData,
             'on_stats' => $onStats,
-        ], $options));
+        ], $options);
+
+        $mergedOptions = array_map(function($mergedOption) {
+            if ( is_array( $mergedOption ) ) {
+                $mergedOption = array_map( function ( $value ) {
+                    return $value instanceof Stringable ? $value->toString() : $value;
+                }, $mergedOption );
+            } else {
+                $mergedOption = $mergedOption instanceof Stringable ? $mergedOption->toString() : $mergedOption;
+            }
+
+            return $mergedOption;
+        }, $mergedOptions);
```

Comment:
Not a gigantic fan of this implementation.

Super happy for someone to come up with a more elegant approach!